lib/ansible/parsing/dataloader.py

```diff
@@ -420,7 +424,7 @@ def get_real_file(self, file_path, decrypt=True):
                         # since the decrypt function doesn't know the file name
                         data = f.read()
                         if not self._b_vault_password:
-                            raise AnsibleParserError("A vault password must be specified to decrypt %s" % file_path)
+                            raise AnsibleParserError("A vault password must be specified to decrypt %s" % to_text(file_path))
```

Comment:
This is where the variable is being handed to the message of an exception constructor.  So it's where to_native() is appropriate.